fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1136,7 +1136,7 @@ ACTOR Future<Void> postResolution(CommitBatchContext* self) {
 	wait(pProxyCommitData->latestLocalCommitBatchLogging.whenAtLeast(localBatchNumber - 1));
 	wait(yield(TaskPriority::ProxyCommitYield1));
 
-	self->computeStart = g_network->now();
+	self->computeStart = g_network->timer();
```

Comment:
Can you describe the problem and how the change fix the problem in the PR? 

I doubt how this fix correctness failures. 
This change makes the deterministic test no longer deterministic. That's why the previous failures are no longer reproducible.

Reply:
The original code was using `timer()`, so this PR reverts previous change in commit a7cff48. I agree we should verify this indeed fix the problem.

Reply:
@xumengpanda : Sim2 makes `timer()` determinism it seems by adding a deterministic random value so should be fine. But yes, I don't understand how it helped, as among all values that `timer()` can return it also includes the `[now(), now() + 0.1]`.

Reply:
@xis19 Can you add an assert in g_network->timer(), checking if `deterministicRandom()->random01()*(time+0.1-timerTime)/2.0;` can be negative?

Reply:
I tried to dig into more details on this issue. I tested MemoryLifetime and also Watches, using the `now` function rather than `timer`. MemoryLifetime will fail on bulkSetup, actor setupRangeWorker, where a few commits would be completed successfully before the final failure. On the other hand, Watches, which is not depending on bulkSetup, would also fail during its initialization process after a few commits done successfully. Both initialization would stuck on infinity loop due to `Transaction is too old to perform reads or be committed` error. With `timer` approach the same error is seen, but the process is not stuck. It is also observed that `deterministicRandom()->random01()*(time+0.1-timerTime)/2.0` is never negative as `timer` is always same or ahead of `now`, by printing out the difference.

Reply:
I guess this is the best extent I could reach at this time. @xumengpanda  -- do you think I should put the assert inside the master branch or my current verification is sufficient?

Reply:
This is something really strange and interesting. 
I think your effort on this is good enough. Since this PR fixes the correctness regression, you can put this problem aside now.

I will put this into my TODO list and do some further investigation.

Reply:
Thanks!